src/dc_tools.rs

```diff
@@ -1069,137 +1084,88 @@ pub unsafe fn dc_get_abs_path(
     pathNfilename_abs
 }
 
-#[allow(non_snake_case)]
-pub unsafe fn dc_file_exist(context: &Context, pathNfilename: *const libc::c_char) -> libc::c_int {
-    let pathNfilename_abs = dc_get_abs_path(context, pathNfilename);
-    if pathNfilename_abs.is_null() {
-        return 0;
+pub fn dc_file_exist(context: &Context, path: *const libc::c_char) -> libc::c_int {
+    if let Some(path_abs) = dc_get_abs_path_safe(context, as_str(path)) {
```

Comment:
A path should not be coerced using `as_str()`, it should be turned into `OsString` or directly to `Path`.

Reply:
Fixed, using `as_path` now